interface/src/app/tribe_client/genesets.js

```diff
@@ -0,0 +1,88 @@
+angular.module('adage.tribe_client.genesets', [
+  'adage.tribe_client.resource'
+])
+
+// Geneset Search Factory
+// Allows for retreiving for and interacting with search results for genesets.
+.factory('GenesetSearch', function($rootScope, Genesets) {
+  var genesets = [];
+  var query = {};
+  var totalResults = 0;
+
+  return {
+    getQuery: function() {
+      return query;
+    },
+    getGenesets: function() {
+      return genesets;
+    },
+    clear: function() {
+      query = {};
+      genesets = [];
+      $rootScope.$broadcast('genesets.update');
+    },
+    totalResults: function() {
+      return totalResults;
+    },
+    query: function(searchParams) {
+      query = searchParams;
+      Genesets.query(query, function(data) {
+        totalResults = data.meta.total_count;
+        genesets = data.objects;
+        $rootScope.$broadcast('genesets.update');
+      });
+    }
+  };
+})
+
+// Search box for users to enter their search text into
+// and go search Tribe.
+.directive('genesetSearchForm', function(GenesetSearch) {
+  return {
+    controller: ['$scope', function($scope) {
+      $scope.searchGenesets = function(search) {
+        search['organism__scientific_name'] = $scope.organism;
+        GenesetSearch.query(search);
+      };
+    }],
+    replace: true,
+    restrict: "E",
+    scope: {
+      limit: '=limit',
+      organism: '='
+    },
```

Comment:
Awesome. You are already using an isolate scope, so this is where you add a parameter for your `results` to get passed back to the parent... ☆